frontend/src/lib/components/PropertyFilters/components/taxonomicPropertyFilterLogic.ts

```diff
@@ -129,9 +133,12 @@ export const taxonomicPropertyFilterLogic = kea<taxonomicPropertyFilterLogicType
                         property_value_type_to_default_operator_override[propertyValueType ?? ''] ||
                         PropertyOperator.Exact
 
+                    const isGroupNameFilter = taxonomicGroup.type.startsWith(TaxonomicFilterGroupType.GroupNamesPrefix)
+                    // :TRICKY: When we have a GroupNamesPrefix taxonomic filter, selecting the group name
+                    // is the equivalent of selecting a property value
                     const property: AnyPropertyFilter = {
-                        key: propertyKey.toString(),
-                        value: null,
+                        key: isGroupNameFilter ? '$group_key' : propertyKey.toString(),
+                        value: isGroupNameFilter ? propertyKey.toString() : null,
```

Comment:
This override is the last thing that raises eyebrows... Happy to let is pass though, the impact is very minimal.

The bigger UX sadness here, which can also be for later, is that once you select an organization, the filter says "Group key", not "Organization key".

![2024-01-19 13 27 07](https://github.com/PostHog/posthog/assets/7115141/9b3f1bb1-d100-47e2-a89a-4c3fd026e765)

The title there can probably be field, but up to you if you want to...

Reply:
Want to keep it is `group_key` for now, given it's in context of matching org(s), and this is the actual field when, say, calling `$groupIdentify()`. Will revisit though if this raises confusion